components/Weather.js

```diff
@@ -1,146 +1,221 @@
 import React from 'react';
-import {View, Text, StyleSheet, Image, TouchableHighlight} from 'react-native';
+import { View, StyleSheet, Alert } from 'react-native';
 import Geolocation from 'react-native-geolocation-service';
+import AsyncStorage from '@react-native-community/async-storage';
+import NetInfo from '@react-native-community/netinfo';
+import PropTypes from 'prop-types';
+import config from '../app.config';
 import Location from './Location';
 import MainPanel from './MainPanel';
 import NextDays from './NextDays';
+import Welcome from './WelcomeScreen';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    flexDirection: 'column',
+    padding: 5,
+    justifyContent: 'center',
+  },
+  baseText: {
+    fontFamily: 'Roboto',
+    color: '#fff',
+    fontSize: 22,
+  },
+  heading: {
+    fontSize: 35,
+  },
+});
+
 export default class Weather extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      latitude: null,
+      longitude: null,
+      cityName: null,
+      weatherFetched: false,
+      phoneLocated: false,
+      weather: {
+        days: [],
+        city: null,
+        currenWeather: null,
+        nearestHours: null,
+      },
+    };
   }
-  state = {
-    latitude: null,
-    longitude: null,
-    cityName: null,
-    weatherFetched: false,
-    phoneLocated: false,
-    weather: {dataPerHour: []},
+
+  static propTypes = {
+    hasLocationPermission: PropTypes.bool,
   };
```

Comment:
Use propsTypes in a way which you have done it in previous components.

Reply:
What do you mean ?